store/policy.go

```diff
@@ -238,8 +312,8 @@ func (s *Store) findPolicyImpl(ctx context.Context, tx *sql.Tx, find *api.Policy
 // upsertPolicyRaw sets a policy for an environment.
 func (s *Store) upsertPolicyRaw(ctx context.Context, upsert *api.PolicyUpsert) (*policyRaw, error) {
 	// Validate policy.
-	if upsert.Type != "" {
-		if err := api.ValidatePolicy(upsert.Type, upsert.Payload); err != nil {
+	if upsert.Type != "" && upsert.Payload != nil {
+		if err := api.ValidatePolicy(upsert.Type, *upsert.Payload); err != nil {
```

Comment:
Not sure if we need upsert.Payload != nil and de-reference here since the argument is a pointer

Reply:
We need to unmarshal the payload and check if is valid in `ValidatePolicy` it. We can also change the `ValidatePolicy` function from `ValidatePolicy(pType PolicyType, payload string)` to `ValidatePolicy(pType PolicyType, payload *string)` and check the nil pointer inside it.

Reply:
oh, I get confused by hasAccessToUpsertPolicy

let's make hasAccessToUpsertPolicy accept non-pointer.

Reply:
Update, PTAL